paxml/tasks_lib.py

```diff
@@ -811,6 +820,14 @@ def get_excluded_var_mask_for_grad_or_opt(
     excluded_for_grad = py_utils.match_variable_names(
         var_weight_hparams, learner.bprop_variable_exclusion
     )
+  if mask_all_overwrite:
+    overwrite_var_mask = \
+        get_overwrite_with_gradient_var_mask(var_weight_hparams)
+    excluded_for_grad = jax.tree_map(
+        lambda is_overwrite, old: is_overwrite or old,
```

Comment:
let's use better variable naming. Other functions are using 'e' for elements of excluded_for_grad

Reply:
Let's follow the pattern and define base_layer.var_overwrite_with_gradient(x) function.
and pass var_weight_hparams to this tree_map directly, just like the tree_map below.

Reply:
Since we still want the old way to provide the var_overwrite_with_gradient from base_layer from praxis, it seems the PRs from paxml and praxis still depend on each other. I will merge the changes and still do two PRs.